### ISSUE TITLE
username containg "." (dot) character not passing schema validation

### ISSUE THREAD

**Issue by amolari** [F5Networks/f5-declarative-onboarding#277]
### Environment
 * Declarative Onboarding Version: 1.24.0
 * BIG-IP Version: 14.1.4

### Summary
defining a user with a name containing a "." (dot) character doesn't pass the schema validation.
User is locally defined on the Big-IP and authentication method is ldap. Username containing a dot character
is supported on TMSH (and yaml)

### Steps To Reproduce
declaration (snippet):
```yaml
          a-john.doe:
            class: User
            userType: regular
            shell: tmsh
            partitionAccess:
              all-partitions:
                role: admin
```

### Expected Behavior
User should be configured.
TMSH supports dot character in username:

```
tmsh list auth user a-john.doe

auth user a-john.doe {
    encrypted-password <redacted>
    partition Common
    partition-access {
        all-partitions {
            role admin
        }
    }
    session-limit -1
    shell tmsh
}
```

### Actual Behavior
```json
[
	{
		"keyword": "pattern",
		"dataPath": ".declaration.Common",
		"schemaPath": "#/properties/Common/propertyNames/pattern",
		"params": {
			"pattern": "^[A-Za-z][0-9A-Za-z_-]{0,47}$"
		},
		"message": "should match pattern \"^[A-Za-z][0-9A-Za-z_-]{0,47}$\"",
		"propertyName": "a-john.doe"
	},
	{
		"keyword": "propertyNames",
		"dataPath": ".declaration.Common",
		"schemaPath": "#/properties/Common/propertyNames",
		"params": {
			"propertyName": "a-john.doe"
		},
		"message": "property name 'a-john.doe' is invalid"
	}
]
```

**Comment by amolari** [comment]
I think this line
https://github.com/F5Networks/f5-declarative-onboarding/blob/3c9c9393b43eeb9b6775628f14be9ac19863e1f6/src/schema/latest/base.schema.json#L132
should be changed to
`"pattern": "^[A-Za-z][0-9A-Za-z_\\-\\.]{0,47}$"`

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-2846.

**Comment by dstokesf5** [comment]
This has been resolved in the 1.27.0 release.
